plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -248,6 +248,9 @@ private static Optional<LastCheckpoint> findCheckpoint(TrinoFileSystem fileSyste
                 }
 
                 long version = checkpoint.get().version();
+                if (version == endVersion) {
+                    return checkpoint;
+                }
```

Comment:
It would be worth to have a test in `TestTransactionLogAccess` for the usecase where the `_delta_log` directory contains some other checkpoints following the requested version (which matches with the checkpoint version).
Before this change, we would have analyzed all the files of the directory.

Reply:
TestTransactionLogAccess doesn't trace the count of file listing, so it's hard to verify this change at this moment. 
Let's handle in https://github.com/trinodb/trino/issues/21366